### PR TITLE
tests: dedupe some CI job names

### PR DESCRIPTION
**Pull request by brendankenny**
Small job name tweaks. Github actions in status checks are prefaced by their workflow name (e.g. `DevTools / web-tests`) but annoyingly are not prefaced in other contexts (so you get only `web-tests`). This is fine for some things, but I already got confused by the new "Smoke batch 1" and "Smoke batch 2" introduced in #13546.

List of passed status checks with confusing overlapping names

This renames them to "Smoke DevTools 1" and "Smoke DevTools 2", which will end up with a redundant "DevTools / Smoke DevTools 1" in some contexts, but I couldn't think of another good way to dedupe when it's on its own.
Also renames "Windows" to "Windows smoke" and "Windows unit".

The Github UI choices are kind of unfortunate. Things are still somewhat confusing, but at least differentiated. Super happy to bikeshed :)

### CODE REVIEW

File: .github/workflows/devtools.yml [hunk 1]
```diff
@@ -125,11 +125,7 @@ jobs:
       # e.g. if set 1 fails, continue with set 2 anyway
       fail-fast: false
     runs-on: macos-latest
-    env:
-      # The total number of shards. Set dynamically when length of single matrix variable is
```
Comment (brendankenny) [inline .github/workflows/devtools.yml:129]: driveby: in this case it can be dynamic, because the matrix size is just the array of shards

File: .github/workflows/devtools.yml [hunk 2]
```diff
@@ -125,11 +125,7 @@ jobs:
...
-      # computable. See https://github.community/t/get-length-of-strategy-matrix-or-get-all-matrix-options/18342
-      SHARD_TOTAL: 2
-    name: Smoke batch ${{ matrix.smoke-test-shard }}
+    name: Smoke DevTools ${{ matrix.smoke-test-shard }}
```
Comment (connorjclark) [inline .github/workflows/devtools.yml:128]: nit: "DevTools smoke" or change the ordering of the other names (Ex: "Smoke windows")

**Review by connorjclark — APPROVED** [review]

File: .github/workflows/devtools.yml (hunk 2 above)
Comment (brendankenny) [reply .github/workflows/devtools.yml:128]: I considered that, but then we end up with `DevTools / DevTools smoke` or `smoke / Smoke Windows` haha. I guess it doesn't really matter and consistency is better :)
